lib/datadog/core/configuration.rb

```diff
@@ -258,8 +258,16 @@ def build_components(settings)
       def replace_components!(settings, old)
         components = Components.new(settings)
 
+        # Carry over state from existing components to the new ones.
+        # Currently, if we already started the remote component (which
+        # happens after a request goes through installed Rack middleware),
+        # we will start the new remote component as well.
+        old_state = {
+          remote_started: old.remote&.started?,
+        }
```

Comment:
@marcotc I added a note here per your comment in the other PR, does it sound OK to you?